### ISSUE TITLE
Same version of nugets NEST and NEST.v5 results in different requests (wrong request in NEST.v5 case)

### ISSUE THREAD

**Issue by narcismadern** [elastic/elasticsearch-net#4512]
Background:

We have a piece of code that makes calls to ElasticSearch to compute search results for our websites, and we also have a (created by us) Nuget that uses ElasticSearch as well to store logs of our software (errors, warnings, information about running time, etc).

We are currently using ElasticSearch 5.4, so we want to upgrade to newest version of ES (i.e. 7.x).

We wanted to first upgrade our ES cluster dedicated to our own logs and, in a second wave, move our ES cluster that we use to make searches for our websites.

We thought that using the package source and nugets from https://www.elastic.co/blog/nest-and-elasticsearch-net-upgrading-your-codebase could work for us, so we could have some ES calls running in version 5.4 and others running in 7.x version, temporarily, until we move everything to ES 7.x.

I started following the instructions on https://www.elastic.co/blog/nest-and-elasticsearch-net-upgrading-your-codebase, and I found that, when installing Nugets Elasticsearch.Net.v5 and NEST.v5 (latest version: 5.7.0-ci20191210T133850), our ES cluster returns an error after sending the request:

`{
  "error" : {
    "root_cause" : [
      {
        "type" : "illegal_argument_exception",
        "reason" : "[cardinality] field doesn't support values of type: START_OBJECT"
      }
    ],
    "type" : "illegal_argument_exception",
    "reason" : "[cardinality] field doesn't support values of type: START_OBJECT"
  },
  "status" : 400
}
`

Then I tried different versions of Elasticsearch.Net.v5 and NEST.v5 (all of them being 5.7.0), but all of them resulting in the same error.
In fact, I took a look at the request generated by NEST, and it has some things that I don't understand at all, like how the aggregations are built:

`"aggs": {
				"<aggregationName>": {
					"cardinality": {
						"field": {
							"cachableExpression": false,
							"name": "<propertyName>"
						}
					}
				}
			}`

instead of:

`"aggs": {
				"<aggregationName>": {
					"cardinality": {
				
								   
						"field": "<propertyName>"
	   
					}
				}
			}`

Then, I just tried to install exactly same version of Nugets Elasticsearch.Net and NEST (latest 5.x version: 5.7.0-ci20191210T133850) from the same repository. I understand the only difference in the case of ".v5" versions is that the using needs to contain "5", so "using NEST5;" rather than "using NEST;", so two different versions could live together.

Surprisingly, the difference seems to be also in the internal code of the nugets, since the request is now accepted by our ES cluster, with the usual JSON structure:

`"aggs": {
				"<aggregationName>": {
					"cardinality": {
				
								   
						"field": "<propertyName>"
	   
					}
				}
			}`

Are we doing something wrong?
We don't understand why "Elasticsearch.Net.v5 and NEST.v5" build a different request (and a wrong one apparently) than "Elasticsearch.Net and NEST" by using the exactly same version on both cases.

I understand that, if we want to have version 5.x and version 7.x living together in our solution, we need to use "Elasticsearch.Net.v5 and NEST.v5". Is that correct?

Hopefully someone can help us with this issue and maybe point out what we are doing wrong.

Thanks in advance

**Comment by russcam** [comment]
_(**Tip:** You can enclose code in code blocks by delimiting with \`\`\` on first and last lines. You can indicate language on first line too, after back ticks e.g. \`\`\`c#)_

> { "field": { "cachableExpression": false, "name": "<propertyName>" } }

`cachableExpression` is serialized but it shouldn't be by the serialization component used in 5.x: 

https://github.com/elastic/elasticsearch-net/blob/80e9b969d3fbb51039bc61c02ba05c31334135f6/src/Nest/CommonAbstractions/Infer/Field/FieldJsonConverter.cs#L14-L24

@narcismadern  can you clarify:

1. When using Elasticsearch.Net.v5 and NEST.v5, are you running against the 5.x Elasticsearch cluster with this version of the client?
2. How are both v5 and 7.x clients configured in the application? Can you show a minimum repro of the setup?

**Comment by narcismadern** [comment]
Hi Russ,

> When using Elasticsearch.Net.v5 and NEST.v5, are you running against the 5.x Elasticsearch cluster with this version of the client?

I have installed Elasticsearch.Net.v5 and NEST.v5 (version 5.7.0-ci20191210T133850) and I'm running against an 5.x ES cluster, indeed.
In contrast, when I use same version (5.7.0-ci20191210T133850) by using Nugets Elasticsearch.Net and NEST from the same repository, the requests are accepted by the same 5.x ES cluster, and the requests are OK. Without more changes.
Thus, my guess is that those nugets are indeed different, because I did not introduce any changes in my code, **only** the "usings" ("using NEST5;" and "using Elasticsearch.Net5;" instead of the usual "using NEST;" and "using Elasticsearch.Net;").

> How are both v5 and 7.x clients configured in the application? Can you show a minimum repro of the setup?

At this moment, I've only tried with one version installed (v5) in my code and, since it did not work, I did not install yet version 7.x of the nugets.
I was only testing (still) that our 5.x ES cluster was working fine when using nugets Elasticsearch.Net.v5 and NEST.v5 (version 5.7.0-ci20191210T133850), which was not the case apparently, because it builds (apparently) wrong requests that our 5.x ES cluster do not understand.

If you still need it, I i'll post here how we configured our project to use Elasticsearch.Net.v5 and NEST.v5.

And thanks for the tip ;-)

Thanks!

**Comment by russcam** [comment]
> If you still need it, I i'll post here how we configured our project to use Elasticsearch.Net.v5 and NEST.v5.

Yes please, that would help. Feel free to link to a GitHub repository too

**Comment by russcam** [comment]
Did you get any further with this, @narcismadern ?

**Comment by narcismadern** [comment]
Hi,

We finally decided to simply upgrade everything we have regarding ES in our code together, instead of trying to do it with two different versions of ES.
We did not find any solution to the problem I was exposing in my previous response here (requests generated by same version of Elasticsearch.Net.v5 and Elasticsearch.Net are different and not compatible), so in the end we decided we would spend more time trying to make it work rather than just upgrade everything at once.

Anyway, thanks for your support.
